### PR TITLE
1214 loading indicator dark mode

### PR DESCRIPTION
**Pull request by GCHQ-Developer-741**
## Summary of the changes
- Removed appearance prop from ic-loading-indicator, adding theme to support dark mode
- Updated storybook, tests and docs with prop changes
- Added styles to components where ic-loading-indicator is used internally (such as ic-toast) as their shadow-doms would prevent the dark mode tokens from reaching the component.

## Related issue
#1214 

### General 

- [x] Changes to docs package checked and committed.
- [x] All acceptance criteria reviewed and met. 

### Testing

- [x] Relevant unit tests and visual regression tests added. 
- [x] Visual testing against Figma component specification completed. 
- [x] Playground stories in React Storybook up to date, with any prop changes and additions addressed. 
- [x] Compare performance of modified components against develop using Performance addon in React Storybook.

### Accessibility 

- [x] Accessibility Insights FastPass performed.
- [x] A11y unit test added and yields no issues.
- [x] A11y plug-in on Storybook yields no issues. 
- [x] Manual screen reader testing performed using NVDA and VoiceOver. 
- [x] Manual keyboard testing for keyboard controls and logical focus order. 
- [x] Correct roles used and ARIA attributes used correctly where required. 
- [x] Logical heading structure is maintained, and the HTML elements used for headings can be changed to fit within the wider page structure. 

### Resize/zoom behaviour 

- [x] Page can be zoomed to 400% with no loss of content. 
- [x] Screen magnifier used with no issues. 
- [x] Text resized to 200% with no loss of content.
- [x] Text spacing increased as per the [WCAG 1.4.12 success criterion](https://www.w3.org/TR/WCAG21/#text-spacing) with no loss of content.

### System modes

- [x] Browser setting 'prefers reduced motion' tested. No animations or motion visible whilst this setting is on. 
- [x] Windows High Contrast mode tested with no loss of content.
- [x] System light and dark mode tested with no loss of content.
- [x] Browser support tested (Chrome, Safari, Firefox and Edge). 

### Testing content extremes

- [x] Min/max content examples tested with no loss of content or overflow. 
- [x] All prop combinations work without issue. 
- [x] Tested for FOUC (Flash of Unstyled Content) in both SSR (Server-Side Rendering) and SSG (Static Site Generation) settings.
- [x] Controlled and uncontrolled input components tested.
- [x] Props/slots can be updated after initial render.

### CODE REVIEW

**Review by ad9242 — APPROVED** [review]

**Review by ad9242 — APPROVED** [review]

**Review by GCHQ-Developer-847 — CHANGES_REQUESTED** [review]
A couple of comments:
1. Looks from the designs like the loading indicator needs a `monochrome` prop

2. Some of the variables used are not the same as in the designs (even though they are the same colour). I think it would be good to use the same ones so everything matches the designs, and I think this would reduce the risk of there being unexpected changes in future if we decide to use a different colour for something (e.g. we may want to change the colour of text, and at the moment it would change the loading indicator because `--ic-color-text-primary` is used)

    ... Basically, I think only use the variables listed in the token table on Figma, e.g. instead of `--ic-color-text-primary` for the "inner color" in dark mode, it would be `--ic-action-light`
   
 Hopefully this makes sense - took a while to get my head around things 😅

**Review by gd2910 — DISMISSED** [review]

**Review by gd2910 — APPROVED** [review]

File: packages/react/src/stories/ic-loading-indicator.stories.mdx [hunk 1]
```diff
@@ -246,8 +247,8 @@ export const defaultArgTypes = {
 <Canvas>
   <Story name="Linear dark" parameters={{ backgrounds: { default: "dark" } }}>
     <div style={{ display: "grid", gap: "50px" }}>
-      <IcLoadingIndicator label="Default" type="linear" dark />
-      <IcLoadingIndicator label="Large" size="large" type="linear" dark />
+      <IcLoadingIndicator label="Default" type="linear" theme="dark" />
+      <IcLoadingIndicator label="Large" size="large" type="linear" theme="dark" />
       <IcLoadingIndicator
```
Comment (GCHQ-Developer-847) [inline packages/react/src/stories/ic-loading-indicator.stories.mdx:252]: I think this third example needs changing to use `theme="dark"` as well

File: packages/web-components/src/components/ic-loading-indicator/ic-loading-indicator.stories.mdx [hunk 2]
```diff
@@ -258,83 +259,87 @@ export const defaultArgTypes = {
...
           label="Default"
-          appearance="light"
+          theme="dark"
         ></ic-loading-indicator>
         <ic-loading-indicator
           label="Large"
           size="large"
-          appearance="light"
+          theme="dark"
+        ></ic-loading-indicator>
```
Comment (GCHQ-Developer-847) [inline packages/web-components/src/components/ic-loading-indicator/ic-loading-indicator.stories.mdx:284]: There is already a default-sized loading indicator on line 276 so this one isn't needed

**Review by GCHQ-Developer-847 — CHANGES_REQUESTED** [review]
Just noticed a couple of small things, but looking great overall :)

**Review by GCHQ-Developer-847 — APPROVED** [review]

**Pull request closed without merging**